pkiinit/pkiInit.go

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/edgexfoundry/security-secret-store/pkiinit/option"
+)
+
+func main() {
+	genOpt := flag.Bool("generate", false, "generate pki assets")
```

Comment:
These flags IMO should have been completely encapsulated in the options class.
Unlike C, os.Args is accessible from anywhere and thus flags should not need to be called from main() and passed in.  The options class as part of its initialization can test each of the options.

Reply:
Although it is possible to put flag into option class, the main() still somehow needs to gets that flag.Bool to be present and that is the way to do the option flags in Golang. The way I put is pretty much standard way everyone does.

Reply:
so, in the executable you will do `./pki-init -generate` for example, and that is the syntax or api flag.Bool to read that argument in Golang.